packages/flutter_tools/lib/src/native_assets.dart

```diff
@@ -415,3 +437,203 @@ Uri buildUriMultiple(Uri projectUri) {
+  final Directory buildDir = fileSystem.directory(buildUri_);
+  if (!await buildDir.exists()) {
+    // CMake requires the folder to exist to do copying.
+    await buildDir.create(recursive: true);
+  }
+  if (await hasNoPackageConfig(buildRunner) || await isDisabledAndNoNativeAssets(buildRunner)) {
+    final Uri nativeAssetsYaml =
+        await writeNativeAssetsYaml(<Asset>[], yamlParentDirectory ?? buildUri_, fileSystem);
+    return (nativeAssetsYaml, <Uri>[]);
+  }
+
+  final BuildMode buildModeCli = nativeAssetsBuildMode(buildMode);
+
+  globals.logger.printTrace('Building native assets for $target $buildModeCli.');
+  final BuildResult result = await buildRunner.build(
+    linkModePreference: LinkModePreference.dynamic,
+    target: target,
+    buildMode: buildModeCli,
+    workingDirectory: projectUri,
+    includeParentEnvironment: true,
+    cCompilerConfig: await buildRunner.cCompilerConfig,
+  );
+  final List<Asset> nativeAssets = result.assets;
+  final Set<Uri> dependencies = result.dependencies.toSet();
+  ensureNoLinkModeStatic(nativeAssets);
+  globals.logger.printTrace('Building native assets for $target done.');
+  final Uri? absolutePath = flutterTester ? buildUri_ : null;
+  final Map<Asset, Asset> assetTargetLocations =
+      _assetTargetLocationsLinuxWindows(nativeAssets, absolutePath);
+  await _copyNativeAssetsLinuxWindows(
+    buildUri_,
+    assetTargetLocations,
+    buildMode,
+    fileSystem,
+  );
+  final Uri nativeAssetsUri = await writeNativeAssetsYaml(
+    assetTargetLocations.values,
+    yamlParentDirectory ?? buildUri_,
+    fileSystem,
+  );
+  return (nativeAssetsUri, dependencies.toList());
+}
+
+Map<Asset, Asset> _assetTargetLocationsLinuxWindows(
+  List<Asset> nativeAssets,
+  Uri? absolutePath,
+) =>
+    <Asset, Asset>{
+      for (final Asset asset in nativeAssets)
+        asset: _targetLocationLinuxWindows(asset, absolutePath),
+    };
+
+Asset _targetLocationLinuxWindows(Asset asset, Uri? absolutePath) {
+  final AssetPath path = asset.path;
+  switch (path) {
+    case AssetSystemPath _:
+    case AssetInExecutable _:
+    case AssetInProcess _:
+      return asset;
+    case AssetAbsolutePath _:
+      final String fileName = path.uri.pathSegments.last;
+      Uri uri;
+      if (absolutePath != null) {
+        // Flutter tester needs full host paths.
+        uri = absolutePath.resolve(fileName);
+      } else {
+        // Flutter Desktop needs "absolute" paths inside the app.
+        // "relative" in the context of native assets would be relative to the
+        // kernel or aot snapshot.
+        uri = Uri(path: fileName);
+      }
+      return asset.copyWith(path: AssetAbsolutePath(uri));
+  }
+  throw Exception('Unsupported asset path type ${path.runtimeType} in asset $asset');
+}
+
+/// Extract the [Target] from a [TargetPlatform].
+Target _getNativeTargetLinuxWindows(build_info.TargetPlatform targetPlatform) {
```

Comment:
Why is this Linux/Windows-specific?

Reply:
What was the answer to this? I'm not sure why a `TargetPlatform`->`Target` mapping function wouldn't cover every supported platform.

Reply:
The target locations in MacOS/iOS are overlapping for multiple architectures, because they are fat binaries.
The target locations in Android are also overlapping, but that is because Android automatically resolves within the `jniLibs/lib/$jniArchDir/`.

Maybe I misunderstood the question. Is your question about something else than the paths we embed in the dill file to do lookup at runtime?

Reply:
Oh, great, yeah GitHub UI is showing your comment on other code, this is probably about the function below.

About the function below. For darwin, ios, and android, we need the DarwinArchs,AndroidArchs information. So for those platforms those environment vars are passed in to the similar function.
For Windows/Linux the architecture is _not_ separate from the target platforms (likely because there is no way to build an app that targets more than one platform). So the information is embedded in the TargetPlatform enum instead of a separate WindowsArchs or LinuxArchs environment variable.